lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -726,6 +726,15 @@ void LLVMToSPIRV::transFPGAFunctionMetadata(SPIRVFunction *BF, Function *F) {
           new SPIRVDecorateFuseLoopsInFunctionINTEL(BF, Depth, Independent));
     }
   }
+  if (MDNode *PreferDSP = F->getMetadata(kSPIR2MD::PreferDSP)) {
+    if (BM->isAllowedToUseExtension(ExtensionID::SPV_INTEL_fpga_dsp_control)) {
+      size_t Mode = getMDOperandAsInt(PreferDSP, 0);
+      MDNode *PropDSPPref = F->getMetadata(kSPIR2MD::PropDSPPref);
+      size_t Propagate = PropDSPPref ? getMDOperandAsInt(PropDSPPref, 0) : 0;
+      BM->addCapability(internal::CapabilityFPGADSPControlINTEL);
```

Comment:
If I understand it correctly, this string isn't needed, since the decoration was already mapped on the capability in lib/SPIRV/libSPIRV/SPIRVEnum.h